eventhubs/client/azure-eventhubs/src/main/java/com/azure/eventhubs/implementation/TokenResourceProvider.java

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.eventhubs.implementation;
+
+import java.util.Locale;
+import java.util.Objects;
+
+/**
+ * Generates the correct resource scope to access Event Hubs resources given the authorization type.
+ */
+class TokenResourceProvider {
+    private static final String TOKEN_AUDIENCE_FORMAT = "amqp://%s/%s";
+    private static final String AZURE_ACTIVE_DIRECTORY_SCOPE = "https://eventhubs.azure.net//.default";
+
+    private final CBSAuthorizationType authorizationType;
+    private final String host;
+
+    TokenResourceProvider(CBSAuthorizationType authorizationType, String host) {
+        Objects.requireNonNull(authorizationType);
+        Objects.requireNonNull(host);
+
+        this.host = host;
+        this.authorizationType = authorizationType;
+    }
+
+    String getResourceString(String resource) {
```

Comment:
Am guessing the `resource` here will have different values based on whether the AMQP link in question is a sender, receiver or management, right?

Reply:
Just curious... any reason why the `getResourceString` shouldn't be a static method on a static class? The output only depends on the auth type, host and resource, the class need not maintain any state.